src/reducers/entryDraft.js

```diff
@@ -73,6 +74,10 @@ const entryDraftReducer = (state = Map(), action) => {
     case REMOVE_ASSET:
       return state.update('mediaFiles', list => list.filterNot(path => path === action.payload));
 
+    case UNPUBLISHED_ENTRY_REGISTER_DEPENDENCY:
+      return state.setIn(['entry', 'data', 'dependencies', action.payload.field],
```

Comment:
The entry data is what gets persisted in the repository - we don't want to add "dependencies" to the frontmatter of the entry and the name could easily class with a field.

Either we should store them in something like `entry.dependencies` or do like the assets and have `dependencies` in the state for entryDraft.